### DISCUSSION TITLE
Enhancing the Notary Selection and Incentivization Mechanism

### DISCUSSION THREAD

**Discussion by hyunmoon** [filecoin-project/FIPs#824] [Enhancements - Governance]
### Abstract:
This proposal introduces adjustments to the notary selection process, sets out clear notary incentives, aims to incorporate automation to aid notaries, and suggests a third-party intervention in unique circumstances, all the while ensuring full transparency and preserving the notary system's essence in Filecoin.

### Motivation:
As the Filecoin network grows, refining the notary system becomes pivotal. Feedback and observations have indicated areas of the system that need betterment.

#### Random Notary Selection with Decline Option:
- **Mechanism:** Notaries are randomly selected for tasks, ensuring a broad spectrum of reviewers.
- **Decline Option:** Notaries can decline tasks if they believe another notary might be more suited based on expertise or regional knowledge. Clear guidelines on suitable reasons for declining will be established.

#### Third-party Intervention:
- **Mechanism:** For applications facing continuous declines, a predefined third-party institution will intervene.
- **Selection Criteria:** Third parties are chosen based on a consensus mechanism within the community and are rotated periodically to prevent centralization.
- **Activation Threshold:** Third-party intervention is initiated only after an application has been declined by a certain number of unique notaries, ensuring it's a last-resort measure.

#### Incentivization of Notaries:
- **Reward Pool Functionality:** Set up a reward pool where distributions are based on:
  - Number of tasks accepted.
  - Adherence to predefined Key Performance Indicators (KPIs).
  - Timeliness in task completion.
- **Review:** The incentive structure is evaluated quarterly to maintain its relevance and prevent undue influence.

#### Automation and Scaling:
- **Assistive Automation:** Identify areas where automation can assist notaries, like notifying notaries of new tasks or providing initial risk assessments.
- **Human Oversight:** All automated processes have a clear human oversight mechanism to ensure decisions are balanced.

#### Feedback Mechanism:
- **Platform:** A dedicated platform will be introduced for the community to share feedback on the notary process.
- **Action Steps:** Feedback is compiled, reviewed, and addressed on a biannual basis.

#### Transparency and Trackability:
- **Mechanism:** All notary-related activities will be logged on a transparent and immutable ledger, accessible to the community. This includes notary assignments, declinations, and approvals.

### Rationale:
Introducing these enhancements ensures the notary system remains resilient, transparent, and adaptive to Filecoin's growth needs.

### Backwards Compatibility:
The enhancements are designed to integrate seamlessly with existing processes, focusing on augmentation rather than overhaul.

### Test Cases:
To be developed based on feedback and specific implementation steps.

### Implementation:
A detailed implementation plan will be charted out post community consensus, incorporating technical and operational aspects.

### Security Considerations:
Every introduced change will undergo comprehensive security scrutiny, with a special focus on maintaining data integrity in the transparent ledger and ensuring third-party interventions remain uncompromised.

**Comment by raghavrmadya** [comment]
Hi @hyunmoon , thanks for starting this discussion. Can I please request you to move this discussion here - https://github.com/filecoin-project/notary-governance

**Comment by hyunmoon** [comment]
https://github.com/filecoin-project/notary-governance/issues/979
